packages/smart-wallet/src/offers.js

```diff
@@ -71,19 +71,28 @@ export const makeOfferExecutor = ({
         status = { ...status, ...changes };
         onStatusChange(status);
       };
+
       /**
        * Notify user and attempt to recover
        *
        * @param {Error} err
+       * @param {UserSeat<unknown>?} seatRef
        */
-      const handleError = err => {
+      const handleError = (err, seatRef) => {
         logger.error('OFFER ERROR:', err);
         updateStatus({ error: err.toString() });
         void paymentsManager.tryReclaimingWithdrawnPayments().then(result => {
           if (result) {
             updateStatus({ result });
           }
         });
+        if (seatRef) {
+          E(seatRef)
+            .tryExit()
+            .catch(e => {
+              logger.error('EXIT OFFER ERROR:', e);
```

Comment:
this isn't an error if the seat was already exited. generally the contracts _should_ do that. This is a backstop.

let's only `tryExit` if it hasn't already exited (chain with the `hasExited()` promise). Then an exception would be an error. A pretty big one though so I think we'd want to throw instead of merely catch and log.